api/app/controllers/spree/api/v1/users_controller.rb

```diff
@@ -5,22 +5,53 @@ class UsersController < Spree::Api::BaseController
         rescue_from Spree::Core::DestroyWithOrdersError, with: :error_during_processing
 
         def index
-          @users = Spree.user_class.accessible_by(current_ability, :show)
+          load_users
 
           if params[:ids]
-            @users = @users.where(id: params[:ids])
-          elsif params[:q]
-            users_with_ship_address = @users.with_address(params[:q][:ship_address_firstname_start])
-            users_with_bill_address = @users.with_address(params[:q][:ship_address_firstname_start], :bill_address)
-
-            users_with_addresses_ids = (users_with_ship_address.ids + users_with_bill_address.ids).compact.uniq
-            @users = @users.with_email_or_addresses_ids(params[:q][:email_start], users_with_addresses_ids)
+            load_users_by_ids
+          elsif params.dig(:q, :ship_address_firstname_start)
+            load_users_by_address
+          elsif params.dig(:q, :email_start)
+            load_users_by_email
           end
 
+          prepare_index_response
+          respond_with(@users)
+        end
+
+        def load_users
+          @users = Spree.user_class.accessible_by(current_ability, :show)
```

Comment:
```suggestion
          @users ||= Spree.user_class.accessible_by(current_ability, :show)
```

you are using this `@users` variable in other places